base/precompilation.jl

```diff
@@ -51,6 +52,32 @@ function ExplicitEnv(envpath::String=Base.active_project())
         end
     end
 
+    # This project might be a package, in that case, that is also a "dependency"
+    # of the project.
+    proj_name = get(project_d, "name", nothing)::Union{String, Nothing}
+    _proj_uuid = get(project_d, "uuid", nothing)::Union{String, Nothing}
+    proj_uuid = _proj_uuid === nothing ? nothing : UUID(_proj_uuid)
+
+    project_is_package = proj_name !== nothing && proj_uuid !== nothing
+    if project_is_package
+        # TODO: Error on missing uuid?
+        project_deps[proj_name] = UUID(proj_uuid)
+        names[UUID(proj_uuid)] = proj_name
+    end
+
+    if !project_is_package
```

Comment:
```suggestion
    else
```

Reply:
I actually kind of want to have these separate because they do sort of different things so it makes sense to me that they are properly separated.